File: src/main/java/com/technews/aggregate/posts/domain/Post.java
package com.technews.aggregate.posts.domain;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Collections;
import java.util.List;

@Document(collection = "posts")
@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Post {
    public static final Post EMPTY =
            new Post("", "", "", "", "", "", "", Collections.EMPTY_LIST, false, "");

    private String id;
    private String subject;
    private String title;
    private String url;

    private String category;
    private String writer;
    private String date;
    List<String> tags;

    private boolean shared;
    private String createdDt;
}

File: src/main/kotlin/com/technews/aggregate/posts/domain/Post.kt
package com.technews.aggregate.posts.domain

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

@Document(collection = "posts")
class Post(
    @Id
    val id: String? = null,
    val subject: String = "",
    val title: String = "",
    val url: String = "",
    val category: String = "",
    val writer: String = "",
    val date: String = "",
    var tags: List<String> = emptyList(),
    var shared: Boolean = false,
    val createdDt: String = "",
) {
    fun share() {
        shared = true
    }
}
